### PR TITLE
fix: Owner change modal not showing content as expected

### PR DESCRIPTION
**Pull request by yash-rajpal**
## Proposed changes (including videos or screenshots)

image

## Further comments

---

This pull request addresses an issue in the RocketChat/Rocket.Chat repository where the Owner change modal was not displaying content as expected. The changes are made in the `ConfirmOwnerChangeModal.tsx` file. The component has been refactored to utilize the `Trans` component from `react-i18next` for translations and the `useEffectEvent` hook for managing side effects. This update replaces the previous implementation that used the `useTranslation` hook and `RawText` component. The source branch for this fix is `delete-user-owner`, and it is targeted to be merged into the `develop` branch.

[CORE-1093](https://rocketchat.atlassian.net/browse/CORE-1093)

[CORE-1093]: https://rocketchat.atlassian.net/browse/CORE-1093?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/meteor/client/components/ConfirmOwnerChangeModal.tsx [hunk 1]
```diff
@@ -19,58 +19,64 @@ const ConfirmOwnerChangeModal = ({
...
+					components={{ bold: <Box is='span' fontWeight='bold' /> }}
+				/>
+			);
+		}
+		if (shouldBeRemoved.length <= 5) {
+			return (
+				<Trans
+					i18nKey='__count__empty_rooms_will_be_removed_automatically__rooms__'
+					values={{ count: shouldBeRemoved.length, rooms: shouldBeRemoved.join(', ') }}
+					components={{ br: <br /> }}
```
Comment (jessicaschelly) [inline apps/meteor/client/components/ConfirmOwnerChangeModal.tsx:67]: should we have the bold here as well?
```suggestion
					components={{ br: <br />, bold: <Box is='span' fontWeight='bold' /> }}
```

File: apps/meteor/tests/e2e/administration.spec.ts [hunk 2]
```diff
@@ -111,6 +112,47 @@ test.describe.parallel('administration', () => {
...
+
+				// TODO: refactor createChannel utility in order to get channel data when creating
+				const response = await api.post('/channels.create', { name: channelName, members: [user.username] });
+				const { channel } = await response.json();
+
+				await api.post('/channels.addOwner', { roomId: channel._id, username: user.username });
+				await api.post('/channels.removeOwner', { roomId: channel._id, userId: Users.admin.data._id });
+			});
+
+			test('expect to show owner change modal', async ({ page }) => {
```
Comment (jessicaschelly) [inline apps/meteor/tests/e2e/administration.spec.ts:138]: Can we add a test to cover the empty rooms?

**Review by jessicaschelly — CHANGES_REQUESTED** [review]

File: apps/meteor/client/components/ConfirmOwnerChangeModal.tsx (hunk 1 above)
Comment (yash-rajpal) [reply apps/meteor/client/components/ConfirmOwnerChangeModal.tsx:67]: Yep, the translation uses bold component.

File: apps/meteor/tests/e2e/administration.spec.ts (hunk 2 above)
Comment (yash-rajpal) [reply apps/meteor/tests/e2e/administration.spec.ts:146]: Done!

**Review by jessicaschelly — DISMISSED** [review]

File: apps/meteor/tests/e2e/administration.spec.ts [hunk 3]
```diff
@@ -111,6 +112,87 @@ test.describe.parallel('administration', () => {
...
+				user = (await createUserResponse.json()).user;
+
+				const createOwnerUserResponse = await api.post('/users.create', {
+					email: faker.internet.email(),
+					name: faker.person.fullName(),
+					password: faker.internet.password(),
+					username: faker.internet.userName(),
+				});
+
+				expect(createOwnerUserResponse.status()).toBe(200);
```
Comment (MarcosSpessatto) [inline apps/meteor/tests/e2e/administration.spec.ts:140]: same

**Review by dougfabris — APPROVED** [review]
LGTM!

**Review by jessicaschelly — APPROVED** [review]

**Review by MarcosSpessatto — APPROVED** [review]
